### PR TITLE
Format the swig script code and add simd support for python api

### PR DESCRIPTION
**Pull request by Presburger**
#238
Signed-off-by: Yusheng.Ma <Yusheng.Ma@zilliz.com>

### CODE REVIEW

File: python/knowhere/knowhere.i [hunk 1]
```diff
@@ -94,73 +96,97 @@ DOWNCAST ( IndexAnnoy )
...
 %apply (int *INPLACE_ARRAY2, int DIM1, int DIM2){(int *ids,int nq_2,int k_2)}
+
+
 %inline %{
 
 
-knowhere::DatasetPtr ArrayToDataSet( float* xb,int nb, int dim){
+knowhere::DatasetPtr
+ArrayToDataSet(float* xb, int nb, int dim) {
     auto ret_ds = std::make_shared<Dataset>();
```
Comment (cydrain) [inline python/knowhere/knowhere.i:106]: can you call GenDataset() directly ?

File: python/knowhere/knowhere.i [hunk 2]
```diff
@@ -94,73 +96,97 @@ DOWNCAST ( IndexAnnoy )
...
 
-void DumpResultDataSet(knowhere::DatasetPtr result, float *dis, int nq_1, int k_1, 
-                       int *ids,int nq_2, int k_2){
-    auto ids_ = result->Get<const int64_t *>("ids");
-    auto dist_ = result->Get<const float *>("distance");
-    assert(nq_1==nq_2);
-    assert(k_1==k_2);
+void
+DumpResultDataSet(const knowhere::DatasetPtr& result, float* dis, int nq_1, int k_1, int* ids, int nq_2, int k_2) {
+    auto ids_ = result->Get<const int64_t*>("ids");
```
Comment (cydrain) [inline python/knowhere/knowhere.i:115]: suggest to use GetDatasetIDs() and GetDatasetDistance()

File: python/knowhere/knowhere.i [hunk 3]
```diff
@@ -94,73 +96,97 @@ DOWNCAST ( IndexAnnoy )
...
 
-knowhere::Config CreateConfig(std::string str){
+knowhere::Config
+CreateConfig(const std::string& str) {
     return knowhere::Config::parse(str);
 }
 
-faiss::BitsetView EmptyBitSetView(){
+void
+SetSimdType(const std::string& str) {
```
Comment (cydrain) [inline python/knowhere/knowhere.i:157]: need sync with Milvus configuration:
  # Valid values: [auto, avx512, avx2, avx, sse4_2] 
when set "avx" or "sse4_2", use SimdType::SSE4_2

File: python/setup.py [hunk 4]
```diff
@@ -24,20 +24,22 @@ def run(self):
 
 
 def CheckCUDA():
-    os_loc = os.path.join(KNOWHERE_ROOT, "cmake_build/knowhere/libknowhere.so")
-    x = os.popen("ldd " + os_loc).read()
+    so_loc = os.path.join(KNOWHERE_ROOT, "cmake_build/knowhere/libknowhere.so")
```
Comment (cydrain) [inline python/setup.py:27]: "lib_path" seems better

File: python/knowhere/knowhere.i (hunk 2 above)
Comment (Presburger) [reply python/knowhere/knowhere.i:115]: reduce c++ head api export in python.

File: python/setup.py [hunk 5]
```diff
@@ -61,11 +63,15 @@ def CheckCUDA():
...
     "-Wl,-rpath,$ORIGIN/../../../",
 ]
+
 SWIG_OPTS = [
     "-c++",
     "-I" + os.path.join(KNOWHERE_ROOT, "knowhere"),
 ]
 
+if is_cuda_ver:
+    SWIG_OPTS.append("-DKNOWHERE_GPU_VERSION=1")
```
Comment (cydrain) [inline python/setup.py:73]: better use:
SWIG_OPTS.append("-DKNOWHERE_GPU_VERSION=ON")

Comment (Presburger) [reply python/setup.py:73]: number is better, ON used in cmake option.

**Pull request merged**